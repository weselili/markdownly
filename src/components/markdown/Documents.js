import React from 'react';
import PropTypes from 'prop-types';
import Document from './Document';

function Documents({ documents }) {

  const documentList = documents.map(document => (
    <li key={document.id}>
      <Document title={document.title}/>
    </li>
  ));

  return (
    <ul>
      {documentList}
    </ul>
  );
}

Documents.propTypes = {
  documents: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    markdown: PropTypes.string.isRequired
  })).isRequired,
};

export default Documents;
