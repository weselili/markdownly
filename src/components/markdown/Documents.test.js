import React from 'react';
import { shallow } from 'enzyme';
import Documents from './Documents';

describe('Documents component', () => {
  it('renders Documents', () => {
    const wrapper = shallow(<Documents titles={[]}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
  
