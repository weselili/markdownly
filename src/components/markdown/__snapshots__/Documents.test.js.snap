// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documents component renders Documents 1`] = `<ul />`;
